Add few pal APIs to pal library

fieryorc opened the pull request:
Add few more APIs to the pal library. These are APIs needed for debugger code. We would like to contribute back and keep our changes minimal. 

@mikem8361, can you review this? Thanks.

dnfclas commented:
Hi __@fieryorc__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla2.dotnetfoundation.org.
    </span>

TTYL, DNFBOT;

mikem8361 commented:
LGTM

mikem8361 commented:
@jkotas or @janvorli do you guys have any problem with adding these APIs?

jkotas commented on src/pal/src/misc/interlock.cpp:
```diff
@@ -269,6 +269,24 @@ InterlockedOr(
 }
 
+LONG
+PALAPI
```
The Windows return type for these is BOOLEAN

janvorli replied:
MSDN says the return type is UCHAR

jkotas commented:
Also, it would be nice to add tests for these.

dnfclas commented:
__@fieryorc__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, DNFBOT;

janvorli commented:
LGTM modulo @jkotas comments.

fieryorc commented:
Can you take a quick look? I have updated the code with code review comments and added tests. I ran the tests manually, and they pass. @mikem8361 , @jkotas, @janvorli

janvorli commented:
LGTM

jkotas commented:
@fieryorc Could you please squash the commits before it is merged?

jkotas commented:
LGTM otherwise

mikem8361 commented:
LGTM

fieryorc commented:
Thanks. Merged (to single commit), Rebased on top, and commited.
